### PR TITLE
64-bit allocator and ioctl emulation

### PR DESCRIPTION
**Pull request by Sonicadvance1**

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
Needs an option introduced to make the Jetson's run times not be absolutely abysmal due to old kernel.
Will introduce that tomorrow.

**Comment by Sonicadvance1** [comment]
This is now ready to get merged. Kernels prior to 4.17 have the fastest startup that they can possibly have now.

**Comment by Sonicadvance1** [comment]
Xavier has an interesting failure case that only reproduces in the environment of old kernel + github-actions runner + tests are stealing the full VMA region.
I'll need to take the Xaviers down to update the kernel which will resolve the problem.

File: External/FEXCore/Source/Interface/Core/JIT/Arm64/ALUOps.cpp [hunk 1]
```diff
@@ -345,7 +344,7 @@ DEF_OP(MulH) {
       sxtw(TMP1, GetReg<RA_64>(Op->Header.Args[0].ID()));
       sxtw(TMP2, GetReg<RA_64>(Op->Header.Args[1].ID()));
       mul(TMP1, TMP1, TMP2);
-      sbfx(GetReg<RA_64>(Node), TMP1, 32, 32);
+      ubfx(GetReg<RA_64>(Node), TMP1, 32, 32);
```
Comment (skmp) [inline External/FEXCore/Source/Interface/Core/JIT/Arm64/ALUOps.cpp:347]: nice catch

File: External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp [hunk 2]
```diff
@@ -262,7 +263,7 @@ namespace {
     Graph->Nodes.resize(NodeCount);
     Graph->VisitedNodePredecessors.clear();
     Graph->AllocData.reset();
-    Graph->AllocData.reset((FEXCore::IR::RegisterAllocationData*)malloc(FEXCore::IR::RegisterAllocationData::Size(NodeCount)));
+    Graph->AllocData.reset((FEXCore::IR::RegisterAllocationData*)FEXCore::Allocator::malloc(FEXCore::IR::RegisterAllocationData::Size(NodeCount * 4)));
```
Comment (skmp) [inline External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp:266]: `* 4`? Why?

File: Source/Tests/ELFCodeLoader2.h [hunk 3]
```diff
@@ -111,7 +111,7 @@ class ELFCodeLoader2 final : public FEXCore::CodeLoader {
     if (Elf.ehdr.e_type == ET_DYN) {
       // needs base address
       auto TotalSize  = CalculateTotalElfSize(Elf.phdrs) + (BrkBase ? BRK_SIZE : 0);
-      LoadBase = (uintptr_t)Mapper(0, TotalSize, PROT_NONE, MAP_ANONYMOUS | MAP_PRIVATE, 0, 0);
+      LoadBase = (uintptr_t)Mapper((void*)0x1'0000, TotalSize, PROT_NONE, MAP_ANONYMOUS | MAP_PRIVATE, 0, 0);
```
Comment (skmp) [inline Source/Tests/ELFCodeLoader2.h:114]: why is this needed?

File: Source/Tests/LinuxSyscalls/x32/Ioctl/steams.inl [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (skmp) [inline Source/Tests/LinuxSyscalls/x32/Ioctl/steams.inl:1]: steams? empty? erratic file?

**Review by skmp — APPROVED** [review]
Looks good overall, see comments

File: Source/Tests/LinuxSyscalls/x32/Ioctl/steams.inl (hunk 4 above)
Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/x32/Ioctl/steams.inl]: Oops, erratic file.

File: External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp]: Oops, debugging left over.

File: Source/Tests/ELFCodeLoader2.h (hunk 3 above)
Comment (Sonicadvance1) [reply Source/Tests/ELFCodeLoader2.h]: Was when I was still tinkering, oops.
